src/store/messagesStore.js

```diff
@@ -1256,30 +1259,60 @@ const actions = {
 	},
 
 	/**
-	 * Posts a simple text message to a room
+	 * Forwards message to a conversation. By default , the message is forwarded to Note to self.
 	 *
 	 * @param {object} context default store context;
 	 * will be forwarded;
 	 * @param {object} data the wrapping object;
+	 * @param {object} data.targetToken the conversation token to where the message will be forwarded;
 	 * @param {object} data.messageToBeForwarded the message object;
 	 */
-	async forwardMessage(context, { messageToBeForwarded }) {
-		const response = await postNewMessage(messageToBeForwarded, { silent: false })
-		return response
-	},
+	async forwardMessage(context, { targetToken, messageToBeForwarded }) {
+		const message = cloneDeep(messageToBeForwarded)
+
+		// when there is no token provided, the message will be forwarded to the Note to self conversation
+		if (!targetToken) {
+			let noteToSelf = context.getters.conversationsList.find(conversation => conversation.type === CONVERSATION.TYPE.NOTE_TO_SELF)
+			// If Note to self doesn't exist, it will be regenerated
+			if (!noteToSelf) {
+				const response = await fetchNoteToSelfConversation()
+				noteToSelf = response.data.ocs.data
+				context.dispatch('addConversation', noteToSelf)
+			}
+			targetToken = noteToSelf.token
+		}
+		// Overwrite with the target conversation token
+		message.token = targetToken
+		if (message.parent) {
+			delete message.parent
+		}
+
+		if (message.message === '{object}' && message.messageParameters.object) {
+			const richObject = message.messageParameters.object
+			const response = await postRichObjectToConversation(
+				targetToken,
+				{
+					objectId: richObject.id,
+					objectType: richObject.type,
+					metaData: JSON.stringify(richObject),
+					referenceId: '',
+				},
+			)
+			return response
+		}
 
-	/**
-	 * Posts a simple text message to a room
-	 *
-	 * @param {object} context default store context;
-	 * will be forwarded;
-	 * @param {object} data the wrapping object;
-	 * @param {string} data.token token of the target conversation
-	 * @param {object} data.richObject the rich object;
-	 */
-	async forwardRichObject(context, { token, richObject }) {
-		const response = await postRichObjectToConversation(token, richObject)
+		// If there are mentions in the message to be forwarded, replace them in the message
+		// text.
+		for (const key in message.messageParameters) {
+			if (key.startsWith('mention')) {
+				const mention = message.messageParameters[key]
+				message.message = message.message.replace(`{${key}}`, `@'${mention.name}'`)
```

Comment:
```suggestion
				message.message = message.message.replace(`{${key}}`, `@"${mention.id}"`)
```
Or even further, check if `mention.id` includes a space character to wrap with double quotes or not

Reply:
it should be `mention.name` as it is the display name. The id might differ and not so human-readable ? ( e.g : user1055

Reply:
If the Id is used, it would be rendered again.

I think the main reason we used the name is that otherwise if you forward an `@all` into a second room, everyone from the second room will be pinged.

However the quotes you added here break the thing anyway already.

```suggestion
				message.message = message.message.replace(`{${key}}`, `@${mention.name}`)
```

Reply:
We can have a look into the handling later, but let's not mix it with this "must merge" feature request.
Keep the previous behaviour for now.

Reply:
ATM it will be rendered in different conversation anyway, if `name === id` or `name.split(' ').includes(id)`.
Should be considered in follow-up:

![image](https://github.com/nextcloud/spreed/assets/93392545/bc260931-6759-4297-890c-9382bb920f97)